From: Sorah Fukumori <her@sorah.jp>
Date: Wed, 27 Dec 2023 06:07:57 +0900
Subject: Test_SyncDefaultGems: Fail when editor run

When something went wrong and git launches editor, and a editor chosen
was terminal-based, a test run unnoticeably hangs.

As we intend editors not to be run with --no-edit, GIT_EDITOR should be
defaulted to `false` so let Git command fails when it attempts to start a
editor. This allows catching such unintentional behaviour in test suite.

(Note: Prior to Git v2.32.0, git cherry-pick doesn't support --no-edit.
https://github.com/git/git/commit/39edfd5cbc4d168db19ec1bc867d78ec7211ec39 )
---
 tool/test/test_sync_default_gems.rb | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/tool/test/test_sync_default_gems.rb b/tool/test/test_sync_default_gems.rb
index 3bf87b3..e64c6c6 100755
--- a/tool/test/test_sync_default_gems.rb
+++ b/tool/test/test_sync_default_gems.rb
@@ -175,13 +175,13 @@ module Test_SyncDefaultGems
       result = nil
       out = capture_process_output_to([STDOUT, STDERR]) do
         Dir.chdir("src") do
-          if editor
-            editor, ENV["GIT_EDITOR"] = ENV["GIT_EDITOR"], editor
-            edit = true
-          end
+          orig_editor = ENV["GIT_EDITOR"]
+          ENV["GIT_EDITOR"] = editor || 'false'
+          edit = true if editor
+
           result = SyncDefaultGems.sync_default_gems_with_commits(@target, commits, edit: edit)
         ensure
-          ENV["GIT_EDITOR"] = editor if edit
+          ENV["GIT_EDITOR"] = orig_editor
         end
       end
       assert_equal(success, result, out)
